### PR TITLE
Added design for deviceShifu-plc4x

### PR DESCRIPTION
**Pull request by BtXin**
**What this PR does / why we need it**:
Design for deviceshifu-plc4x
**Will this PR make the community happier**? 
Yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [ ] unit test
- [ ] e2e test
- [X] other (please specify)
Design doc, don't need specific test

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: docs/design/deviceshifu/design-deviceShifu-plc4x.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	 case S7:
+		 drviers.RegsiterS7(drvierManager)
+		 ...
+```
+
+### Serving requests
+deviceShifu-PLC4X would take RESTful-style requests just as other deviceShifu do. 
+PLC4X supports both `read` and `write` requests. 
+
+For read, the method signature looks like:
```
Comment (MrLeea-13155bc) [inline docs/design/deviceshifu/design-deviceShifu-plc4x.md:43]: Do we need to support multiple requests being sent at the same time? like AddQuery().AddQuery()....

Comment (MrLeea-13155bc) [inline docs/design/deviceshifu/design-deviceShifu-plc4x.md:43]: Do we need to support multiple requests being sent at the same time? like AddQuery().AddQuery()....

Comment (BtXin) [reply docs/design/deviceshifu/design-deviceShifu-plc4x.md:43]: I suppose we can.
With multiple `?` as the delimiter, and take whatever comes after the `?` as the query body.

File: docs/design/deviceshifu/design-deviceShifu-plc4x.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# DeviceShifu PLC4X Design
+
+DeviceShifu-PLC4X allows shifu utilize [apache plc4x](https://plc4x.apache.org/) project to integrate PLC devices with various protocols such as S7, Modbus, etc.
+
+## Goal
+Create a deviceShifu-PLC4X type using PLC4X library and allow user to connect PLC devices easily
```
Comment (kris21he) [inline docs/design/deviceshifu/design-deviceShifu-plc4x.md:6]: what's the value of this goal? or what's difference we can provide after implement this feature?
maybe we can make it more explicitly about the word of "easy"?

File: docs/design/deviceshifu/design-deviceShifu-plc4x.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+e.g: http://device-plc/write?holding-register:26:REAL=2.7182818284?holding-register:3:REAL=3.141592657
+```
+The `read` , `write` after `http://device-plc/` will be served as a instruction router.
+`?` would be served as query delimiter, whatever comes between `?` will be seen as the query.
+
+For `read`, the query would only contain the fields requests reading from. 
+For `write`, the query would contain the fields requests trying to write and the value it tries to write. 
+
+## Testing Plan
+We can use existing mock-plc device, create a deviceShifu-PLC4X image add a e2e test to current azure-pipeline to run test against deviceShifu-PLC4X.
```
Comment (kris21he) [inline docs/design/deviceshifu/design-deviceShifu-plc4x.md:76]: do we have defined successful path, some failure paths.
and any edge case need to consider?

Comment (BtXin) [reply docs/design/deviceshifu/design-deviceShifu-plc4x.md:76]: I think that's more of implementation details, which I will leave it to @MrLeea-13155bc to decide when he implements it.

File: docs/design/deviceshifu/design-deviceShifu-plc4x.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+type ProtocolSettings struct {
+	MQTTSetting   *MQTTSetting   `json:"MQTTSetting,omitempty"`
+	OPCUASetting  *OPCUASetting  `json:"OPCUASetting,omitempty"`
+	SocketSetting *SocketSetting `json:"SocketSetting,omitempty"`
++	PLC4XSetting  *PLC4XSetting  `json:"PLC4XSetting,omitempty"`
+}
+
+// PLC4XSetting defines PLC4X specific settings when connecting to an EdgeDevice
+type PLC4XSetting struct {
+    DriverType *string `json:"driverType,omitempty"`
```
Comment (tomqin93) [inline docs/design/deviceshifu/design-deviceShifu-plc4x.md:28]: Should we use custom `type DriverType` instead of string type?

**Review by tomqin93 — COMMENTED** [review]
@BtXin added 1 comment, the rest looks good

**Review by tomqin93 — APPROVED** [review]
LGTM
